test/distributed/_composable/fsdp/test_fully_shard_compile.py

```diff
@@ -382,51 +403,81 @@ def input_creation_fn():
 
         return model_init_fn, input_creation_fn
 
+    def _scaled_dot_product_attention_with_graph_break(
+        self, orig_fn, fullgraph, *args, **kwargs
+    ):
+        if not fullgraph:
+            torch._dynamo.graph_break()
+        return orig_fn(*args, **kwargs)
+
+    def _mock_sdpa(self, fullgraph):
```

Comment:
nit: maybe the name should imply that you're just conditionally adding a graph break to sdpa? `with _maybe_add_graph_break_to_sdpa(...)`